### PR TITLE
Add filters

### PR DESCRIPTION
**Pull request by mikker**
Includes 2 default enabled filters:

* Stripping secrets from headers
* Stripping request bodies

### CODE REVIEW

**Comment by mikker** [comment]
@simitt, I'd love a review but I understand if other feature freeze related stuff comes first.

File: lib/elastic_apm/agent.rb [hunk 1]
```diff
@@ -154,6 +157,10 @@ def set_user(*args)
       instrumenter.set_user(*args)
     end
 
+    def add_filter(key, callback)
+      @http.filters.add(key, callback)
```
[the remark this answers is not in the document]
Comment (mikker) [reply lib/elastic_apm/agent.rb]: The `http` here is the API object that I use to send payloads.

File: lib/elastic_apm/util.rb [hunk 2]
```diff
@@ -18,3 +18,4 @@ def self.inspect_transaction(transaction)
 end
 
 require 'elastic_apm/util/inspector'
+require 'elastic_apm/util/inspector'
```
[the remark this answers is not in the document]
Comment (mikker) [reply lib/elastic_apm/util.rb]: Good catch!

File: lib/elastic_apm/agent.rb [hunk 3]
```diff
@@ -45,11 +45,13 @@ def self.running?
       !!@instance
     end
 
+    # rubocop:disable Metrics/MethodLength
```
[the remark this answers is not in the document]
Comment (mikker) [reply lib/elastic_apm/agent.rb]: I have already disabled some things. I actually like the MethodLength one although I end up disabling it quite often. I like how it forces me to take an active decision about this method being _"too"_ long.

[the remark this answers is not in the document]
Comment (simitt) [reply lib/elastic_apm/agent.rb]: Oh I didn't mean to disable it but to maybe change the default value for how long a method can be.

**Comment by mikker** [comment]
AFAIR the python agent has support for ignoring specific urls. I might do something like that, yeah. Thanks again!

File: lib/elastic_apm/agent.rb (hunk 3 above)
[the remark this answers is not in the document]
Comment (mikker) [reply lib/elastic_apm/agent.rb]: Oh. Yeah, I'll consider that 👍

**Pull request merged**